src/components/List.js

```diff
@@ -4,33 +4,40 @@ import styled from 'styled-components';
 import ActivityCard from './ActivityCard';
 
 export default function List({
-  activitiesOfSelectedFriend,
+  selectedFriendsActivity,
   errorMessage,
   onDeleteActivity,
+  currentFilter,
 }) {
   const activitiesEndRef = useRef(null);
   const scrollToBottom = () => {
     activitiesEndRef.current.scrollIntoView({ behavior: 'smooth' });
   };
 
-  useEffect(scrollToBottom, [activitiesOfSelectedFriend]);
+  useEffect(scrollToBottom, [selectedFriendsActivity]);
+
   return (
     <ListStyle role="list" title="list of activities">
-      {activitiesOfSelectedFriend.map(activity => (
-        <li key={activity.id}>
-          <ActivityCard
-            onDeleteActivity={() => onDeleteActivity(activity.id)}
-            activity={activity.activity}
-            id={activity.id}
-            nameOfSelectedFriend={activity.friend}
-            nameOfSelectedActivity={activity.activity}
-            errorMessage={
-              errorMessage &&
-              `unfortunately something went wrong with your data.`
-            }
-          />
-        </li>
-      ))}
+      {selectedFriendsActivity
+        .filter(
+          activity =>
+            activity.category.includes(currentFilter) || currentFilter === 'all'
+        )
+        .map(activity => (
+          <li key={activity.id}>
+            <ActivityCard
+              onDeleteActivity={() => onDeleteActivity(activity.id)}
+              activity={activity.activity}
+              id={activity.id}
+              nameOfSelectedFriend={activity.friend}
+              nameOfSelectedActivity={activity.activity}
+              errorMessage={
+                errorMessage &&
+                `unfortunately something went wrong with your data.`
```

Comment:
maybe you should move the condition into the ActivityCard and hand over the error message in any case.